pkg/services/sqlstore/dashboard.go

```diff
@@ -374,24 +374,48 @@ func deleteDashboard(cmd *models.DeleteDashboardCommand, sess *DBSession) error
 		"DELETE FROM dashboard_version WHERE dashboard_id = ?",
 		"DELETE FROM annotation WHERE dashboard_id = ?",
 		"DELETE FROM dashboard_provisioning WHERE dashboard_id = ?",
+		"DELETE FROM dashboard_acl WHERE dashboard_id = ?",
 	}
 
 	if dashboard.IsFolder {
-		deletes = append(deletes, "DELETE FROM dashboard_provisioning WHERE dashboard_id in (select id from dashboard where folder_id = ?)")
 		deletes = append(deletes, "DELETE FROM dashboard WHERE folder_id = ?")
 
 		dashIds := []struct {
 			Id int64
 		}{}
-		err := sess.SQL("select id from dashboard where folder_id = ?", dashboard.Id).Find(&dashIds)
+		err := sess.SQL("SELECT id FROM dashboard WHERE folder_id = ?", dashboard.Id).Find(&dashIds)
 		if err != nil {
 			return err
 		}
 
-		for _, id := range dashIds {
+		dashIdsArray := make([]interface{}, len(dashIds))
+		queryEnd := "("
+		for i, id := range dashIds {
 			if err := deleteAlertDefinition(id.Id, sess); err != nil {
 				return err
 			}
+			dashIdsArray[i] = id.Id
+			queryEnd += "?,"
+		}
+		queryEnd = queryEnd[:len(queryEnd)-1] + ")"
+
+		if len(dashIds) > 0 {
+			childrenDeletes := []string{
+				"DELETE FROM dashboard_tag WHERE dashboard_id IN ",
+				"DELETE FROM star WHERE dashboard_id IN ",
+				"DELETE FROM dashboard_version WHERE dashboard_id IN ",
+				"DELETE FROM annotation WHERE dashboard_id IN ",
+				"DELETE FROM dashboard_provisioning WHERE dashboard_id IN ",
+				"DELETE FROM dashboard_acl WHERE dashboard_id IN ",
+			}
+			for _, sql := range childrenDeletes {
+				sqlOrArgs := append([]interface{}{sql + queryEnd}, dashIdsArray...)
```

Comment:
I would suggest to do `delete from <table> where  dashboard_id in(select id from dashboard where folder_id=?)` instead of concatenating strings.

Reply:
It's more readable for sure but I was trying to avoid doing the subquery `select id from dashboard where folder_id=?` too many times as the `dashboard` table can be quite big and the `folder_id` column is not indexed. What do you think?

Reply:
I was about to suggest using `strings.Join()` for building the query placeholders but after reading @marefr's suggestion I'm in favour of it. You are right that there is no index for folder_id but there is a composite one for org_id, folder_id and title so you could use this one alternatively (and fixing the rest queries using the folder_id). WDYT?

Reply:
Makes sense to me.

Reply:
Did the update to use the composite index, thanks for the tip!